gateway/sds_gateway/api_methods/cog_pipelines.py

```diff
@@ -0,0 +1,654 @@
+                    capture, capture_files, temp_path
+                )
+
+                if not reconstructed_path:
+                    error_msg = "Failed to reconstruct DigitalRF directory structure"
+                    processed_data.mark_processing_failed(error_msg)
+                    raise ValueError(error_msg)  # noqa: TRY301
+
+                # Process the waterfall data in JSON format
+                waterfall_result = convert_drf_to_waterfall_json(
+                    reconstructed_path,
+                    capture.channel,
+                    ProcessingType.Waterfall.value,
+                    max_slices=max_slices,
+                )
+
+                if waterfall_result["status"] != "success":
+                    processed_data.mark_processing_failed(waterfall_result["message"])
+                    raise ValueError(waterfall_result["message"])  # noqa: TRY301
+
+                # Create a temporary JSON file
+                import json
+
+                with tempfile.NamedTemporaryFile(
+                    mode="w", suffix=".json", delete=False
+                ) as temp_file:
+                    json.dump(waterfall_result["json_data"], temp_file, indent=2)
+                    temp_file_path = temp_file.name
+                    logger.info(f"Created temporary JSON file at {temp_file_path}")
+
+                try:
+                    # Store the JSON file
+                    new_filename = f"waterfall_{capture_uuid}.json"
+                    store_result = store_processed_data(
+                        capture_uuid,
+                        ProcessingType.Waterfall.value,
+                        temp_file_path,
+                        new_filename,
+                        waterfall_result["metadata"],
+                    )
+
+                    if store_result["status"] != "success":
+                        processed_data.mark_processing_failed(store_result["message"])
+                        raise ValueError(store_result["message"])
+
+                    # Mark processing as completed
+                    processed_data.mark_processing_completed()
+
+                    logger.info(
+                        f"Completed waterfall processing for capture {capture_uuid}"
+                    )
+                    return {
+                        "status": "success",
+                        "capture_uuid": capture_uuid,
+                        "message": (
+                            "Waterfall JSON data processed and stored successfully"
+                        ),
+                        "json_data": waterfall_result["json_data"],
+                        "metadata": waterfall_result["metadata"],
+                        "store_result": store_result,
+                    }
+
+                finally:
+                    # Clean up temporary file
+                    if "temp_file_path" in locals():
+                        Path(temp_file_path).unlink()
```

Comment:
could use a `missing_ok=True` in `unlink()`

https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink